src/Compilers/CSharp/Test/Semantic/Semantics/PropertyAccessTests.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.Semantics
+{
+    public class PropertyAccessTests : CompilingTestBase
+    {
+        [Fact]
+        [Trait("foo", "bar")]
+        public void SetArrayLength()
+        {
+            var text =
+@"class Test
+{
+    void M()
+    {
+        int[] arr = { 1 };
+        arr.Length = 0;
+    }
+}";
+            var comp = CreateCompilation(
+                trees: new[] { Parse(text) },
+                references: new[] { AacorlibRef });
```

Comment:
Can we change this accessor to compile the .cs file on the fly (and delete the .dll altogether)?  Would be good to avoid checking in another copy of the binary...